pkg/apis/vpc/v1alpha1/ipaddressallocation_types.go

```diff
@@ -3,15 +3,14 @@
 
 package v1alpha1
 
-import (
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-)
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 type IPAddressVisibility string
 
-const (
-	IPAddressVisibilityExternal = "EXTERNAL"
-	IPAddressVisibilityPrivate  = "PRIVATE"
+var (
+	IPAddressVisibilityExternal   IPAddressVisibility = "External"
+	IPAddressVisibilityPrivate    IPAddressVisibility = "Private"
+	IPAddressVisibilityPrivateTGW IPAddressVisibility = "PrivateTGW"
```

Comment:
I just recall for PrivateTGW, we need to convert to NSX type Private_TGW
https://gitlab.eng.vmware.com/core-build/nsx/-/blob/nsx-magnus/mp/policy/policy-api/src/main/resources/api_spec/root/nsx-policy/policy-vpc-subnet-ipam.yml#L49
Like what we did for Subnet access mode:
https://github.com/vmware-tanzu/nsx-operator/blob/4e4decd56d10720d662eb4e23c78e2eeec65a363/pkg/nsx/services/subnet/builder.go#L49